app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/labtest/LabTestDetailFragment.kt

```diff
@@ -44,67 +45,99 @@ class LabTestDetailFragment : BaseRecordDetailFragment(R.layout.fragment_lab_tes
-        observeUiState()
         observePdfData()
-        initComments()
+        observeCommentsSyncCompletion()
     }
 
-    override fun getCommentView(): AddCommentLayout = binding.comment
+    override fun onResume() {
+        super.onResume()
+        viewModel.getLabTestDetails(args.labOrderId)
+    }
 
-    override fun getScrollableView() = binding.rvLabTestDetailList
+    @Composable
+    override fun GetComposableLayout() {
 
-    override fun getCommentEntryTypeCode() = CommentEntryTypeCode.LAB_RESULTS
+        val uiState = viewModel.uiState.collectAsState().value
 
-    override fun getParentEntryId(): String? = viewModel.uiState.value.parentEntryId
+        var commentState: CommentsUiState? = null
 
-    override fun getProgressBar(): View = binding.progressBar
+        if (BuildConfig.FLAG_ADD_COMMENTS) {
+            uiState.parentEntryId?.let { commentsViewModel.getComments(it) }
+            commentState = commentsViewModel.uiState.collectAsState().value
+        }
 
-    private fun setUpRecyclerView(hdid: String?) {
-        labTestDetailAdapter = LabTestDetailAdapter(
-            viewPdfClickListener = { viewModel.getLabTestPdf(hdid) }
-        )
+        MyHealthScaffold(
+            title = uiState.toolbarTitle,
+            isLoading = uiState.onLoading,
+            navigationAction = { findNavController().popBackStack() }
+        ) {
+            LabTestScreen(
+                uiState = uiState,
+                onClickViewPdf = { viewModel.getLabTestPdf(args.hdid) },
+                onClickFaq = { context?.redirect(getString(R.string.faq_link)) },
+                onClickComments = ::navigateToComments,
+                commentsSummary = commentState?.commentsSummary,
+                onSubmitComment = ::onSubmitComment,
+            )
+        }
+        handledServiceDown(uiState)
 
-        concatAdapter = ConcatAdapter(labTestDetailAdapter, getRecordCommentsAdapter())
+        if (uiState.onError) {
+            showError()
+            viewModel.resetUiState()
+        }
 
-        binding.rvLabTestDetailList.adapter = concatAdapter
-    }
+        handlePdfDownload(uiState)
 
-    private fun observeUiState() {
-        viewModel.uiState.collectOnStart { state ->
-            binding.progressBar.isVisible = state.onLoading
+        handleNoInternetConnection(uiState)
 
-            handledServiceDown(state)
+        if (commentState?.isBcscSessionActive == false) findNavController().popBackStack()
+    }
 
-            if (state.labTestDetails?.isNotEmpty() == true) {
-                labTestDetailAdapter.submitList(state.labTestDetails)
-                setupComposeToolbar(binding.composeToolbar.root, state.toolbarTitle)
-            }
+    private fun observeCommentsSyncCompletion() {
+        if (BuildConfig.FLAG_ADD_COMMENTS.not()) return
 
-            if (state.onError) {
-                showError()
-                viewModel.resetUiState()
+        observeWork(SYNC_COMMENTS) {
```

Comment:
Should start observing worker from composable scree.
This will help to reduce usage of fragment and migration to compose easier.